index.bs

```diff
@@ -5069,96 +5069,125 @@ No [=extended attributes=] are applicable to dictionaries.
 
 <h3 id="idl-exceptions">Exceptions</h3>
 
-An <dfn id="dfn-exception" export>exception</dfn> is a type of object that
-represents an error and which can be thrown or treated as a first
-class value by implementations.  Web IDL does not allow exceptions
-to be defined, but instead has a number of pre-defined exceptions
-that specifications can reference and throw in their definition of
-operations, attributes, and so on.  Exceptions have an
-<dfn id="dfn-exception-error-name" for="exception" export>error name</dfn>,
-a {{DOMString}},
-which is the type of error the exception represents, and a
-<dfn id="dfn-exception-message" for="exception" export>message</dfn>, which is an optional,
-user agent-defined value that provides human readable details of the error.
-
-There are two kinds of exceptions available to be thrown from specifications.
-The first is a <dfn id="dfn-simple-exception" export>simple exception</dfn>, which
-is identified by one of the following types:
+An <dfn id="dfn-exception" export>exception</dfn> is a type of object that represents an error and
+which can be thrown or treated as a first class value by implementations. Web IDL has a number of
+pre-defined exceptions that specifications can reference and throw in their definition of
+operations, attributes, and so on. Custom exception types can also be defined, as [=interfaces=]
+that [=interface/inherit=] from {{DOMException}}.
+
+In addition to their type, which is either one of the [=simple exceptions=], {{DOMException}}, or a
+derived interface of {{DOMException}}, all exceptions have a
+<dfn id="dfn-exception-message" for="exception" export>message</dfn>, which is an
+[=implementation-defined=] [=string=] that provides human readable details of the error.
+{{DOMException}} instances also have an
+<dfn id="dfn-exception-error-name" for="exception" export>error name</dfn> [=string=], which
```

Comment:
It's weird that we have error name here, but then for `DOMException` we also define an internal name field. I don't think we need both? Message seems similarly duplicated.

Reply:
The distinction is that these are defining the general concepts/types, and the DOMExceptions definitions are fields.

I think we could eliminate "error name", especially since in this PR I narrow it down to be DOMException-specific. I'll try that.

Eliminating "message" is a bit harder since it applies to all exceptions, not just DOMException. But I think it's doable.